spec/support/acceptance/start_mock_servicenow_instance.sh

```diff
@@ -8,34 +8,77 @@ function cleanup() {
 }
 trap cleanup EXIT
 
-rep=$(curl -s --unix-socket /var/run/docker.sock http://ping > /dev/null)
-status=$?
+function start_servicenow() {
+  id=`docker ps -q -f name=mock_servicenow_instance -f status=running`
 
-if [ "$status" == "7" ]; then
-    apt-get -qq update -y 1>&- 2>&-
-    apt-get install -qq docker.io -y 1>&- 2>&-
-fi
+  if [ ! -z "$id" ]
+  then
+    echo "Killing the current mock ServiceNow container (id = ${id}) ..."
+    docker rm --force ${id}
+  fi
 
-set -e
+  docker-compose -f /tmp/servicenow/docker-compose.yml up -d --remove-orphans
+  #docker build /tmp/servicenow -t mock_servicenow_instance
+  #docker run -d --rm -p 1080:1080 --name mock_servicenow_instance mock_servicenow_instance 1>&- 2>&-
 
-id=`docker ps -q -f name=mock_servicenow_instance -f status=running`
+  id=`docker ps -q -f name=mock_servicenow_instance -f status=running`
 
-if [ ! -z "$id" ]
-then
-  echo "Killing the current mock ServiceNow container (id = ${id}) ..."
-  docker rm --force ${id}
-fi
+  if [ -z "$id" ]
+  then
+    echo 'Mock ServiceNow container start failed.'
+    exit 1
+  fi
+  echo 'Mock ServiceNow container start succeeded.'
+}
+
+function yum_install_docker() {
+  yum install -y yum-utils
+  yum-config-manager \
+    --add-repo \
+    https://download.docker.com/linux/centos/docker-ce.repo
+  yum install docker-ce docker-ce-cli containerd.io -y
+  systemctl start docker
 
-docker build /tmp/servicenow -t mock_servicenow_instance
-docker run -d --rm -p 1080:1080 --name mock_servicenow_instance mock_servicenow_instance 1>&- 2>&-
+  curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+  mv /usr/local/bin/docker-compose /usr/bin/docker-compose
+  chmod +x /usr/bin/docker-compose
+}
 
-id=`docker ps -q -f name=mock_servicenow_instance -f status=running`
+function compose_starting() {
+  docker ps --all | grep starting
+}
 
-if [ -z "$id" ]
+function wait_for_compose() {
+  while [ 1 ]
+  do
+    if [ -z "$(compose_starting)" ]
+    then
+      docker ps --all
+      exit
+    fi
+  done
+}
+
+YUM=$(cat /etc/*-release | grep 'CentOS\|rhel')
+
+nodocker=$(which docker 2>&1 | grep "no docker")
+status=$?
+
+if [ ! -z "$nodocker" ]
 then
-  echo 'Mock ServiceNow container start failed.'
-  exit 1
+  if [ ! -z "$YUM" ]; then
+    yum_install_docker
+  fi
+else
+  apt-get -qq update -y 1>&- 2>&-
+  apt-get install -qq docker.io -y 1>&- 2>&-
+  apt-get install -qq docker-compose -y 1>&- 2>&-
 fi
```

Comment:
I think I don't fully understand the logic on this block. If docker is not installed then check if the system is using yum and install it, but if it is installed then go ahead and install it and compose with apt?

Reply:
It's unclear to me as well. I think I remember from Splunk that it's there to update via apt-get if you're on Ubuntu and you're docker already exists.